homeassistant/components/homekit/util.py

```diff
@@ -44,3 +45,21 @@ def show_setup_message(bridge, hass):
 def dismiss_setup_message(hass):
     """Dismiss persistent notification and remove QR code."""
     hass.components.persistent_notification.dismiss(HOMEKIT_NOTIFY_ID)
+
+
+def convert_to_float(state):
+    """Return float of state, catch errors."""
+    try:
+        return float(state)
+    except (ValueError, TypeError):
+        return None
+
+
+def temperature_to_homekit(temperature, unit):
+    """Convert temperature to Celsius for HomeKit."""
+    return round(temp_util.convert(temperature, unit, TEMP_CELSIUS), 1)
+
+
+def temperature_to_states(temperature, unit):
+    """Convert temperature back form Celsius to Home Assistant unit."""
```

Comment:
1. Type "form"
2. https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/temperature.py

Reply:
I want to preserve the original unit of the climate device. That's why I only use `util.temperature.convert`.

Reply:
Ok, good 👍